daemon/daemon.go

```diff
@@ -125,13 +124,7 @@ func (d *Daemon) UpdateProxyRedirect(e *endpoint.Endpoint, l4 *policy.L4Filter)
 		return 0, fmt.Errorf("can't redirect, proxy disabled")
 	}
 
-	// proxyKind only has an effects on newly created redirects
-	proxyKind := proxy.ProxyKindOxy
-	if viper.GetBool("envoy-proxy") {
-		proxyKind = proxy.ProxyKindEnvoy
-	}
-
-	r, err := d.l7Proxy.CreateOrUpdateRedirect(l4, e.ProxyID(l4), e, proxyKind)
+	r, err := d.l7Proxy.CreateOrUpdateRedirect(l4, e.ProxyID(l4), e)
```

Comment:
The previous approach was a bit slower as it actually read the environment whenever a redirect was created. This would have enabled switching between proxies without restarting cilium-agent, which could be useful for CI testing, i.e., running tests with HTTP redirects twice, once with Oxy and once with Envoy. This was we could get rid of the Envoy-specific parallel Jenkins step and reduce stress on Jenkins.